File: Web/src/client/rest/notifications/NotificationClient.ts
import { Notification } from '../../../api';
import { http } from '../http';

export const notificationClient = {
    getNotifications(): Promise<Notification[]> {
        return http.get<Notification[]>("/notifications");
    },
    acknowledge(id: string): Promise<void> {
        return http.put<void>(`/notifications/${id}/acknowledge`);
    },
}

File: XOutput.Client/Rest/Notifications/NotificationClient.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace XOutput.Rest.Notifications
{
    public class NotificationClient : HttpJsonClient
    {
        public NotificationClient(IHttpClientProvider clientProvider) : base(clientProvider)
        {

        }

        public Task<IEnumerable<Notification>> GetNotificationsAsync(int timeoutMillis = 1000)
        {
            return GetAsync<IEnumerable<Notification>>("api/notifications", CreateToken(timeoutMillis));
        }

        public Task AcknowledgeAsync(string id, int timeoutMillis = 1000)
        {
            return PutAsync("api/notifications/" + id + "/acknowledge", CreateToken(timeoutMillis));
        }
    }
}
